doc/development_guide/how_tos/custom_checkers.rst

```diff
@@ -218,11 +218,26 @@ Now we can debug our checker!
 .. Note::
 
     ``my_plugin`` refers to a module called ``my_plugin.py``.
-    This module can be made available to pylint by putting this
-    module's parent directory in your ``PYTHONPATH``
+    The preferred way of making this plugin available to pylint is
+    by installing it as a package, such as with ``pip install .``.
+
+    Alternatively, this module can be made available to pylint by
+    putting this module's parent directory in your ``PYTHONPATH``
     environment variable or by adding the ``my_plugin.py``
     file to the ``pylint/checkers`` directory if running from source.
```

Comment:
```suggestion
    environment variable.
```

Let's no longer recommend to hack into the `pylint` source. I only foresee additional issues. 😅

Reply:
Yes, I was a little surprised to see that in there! It's worth noting that I, as so many engineers do, had gone to stack overflow to see what I might be missing. I found one other person complaining about something similar, and they solved it but putting their module in the checkers director 😱 

I left an answer there, linking #7264. I've been updating it as my understanding grows, so hopefully anyone else finding that question will realise that putting things in the pylint source is "not considered best practice" 😄 

https://stackoverflow.com/questions/71590677/cant-load-custom-pylint-module/73250367#73250367